src/Execution/lbsCriterionBase.py

```diff
@@ -46,78 +46,50 @@
 import sys
 import bcolors
 
-from src.Model.lbsRank import Rank
-
+from src.Model.lbsWorkModelBase import WorkModelBase
 
 class CriterionBase:
     __metaclass__ = abc.ABCMeta
     """An abstract base class of optimization criteria for LBAF execution
     """
 
-    def __init__(self, ranks, edges, parameters=None):
+    def __init__(self, work_model, parameters: dict=None):
```

Comment:
Ok, so now whay do we need `parameters` in this Class if it's not used?

Reply:
It is currently used by **some** of the concrete subclasses and in the future might become necessary for *all**